tests/ctst/common/common.ts

```diff
@@ -300,40 +300,58 @@ Then('i {string} be able to add user metadata to object {string}',
-                // or that it didn't need to change because there was no new messages
-                assert.ok(newOffsets[i].partitions[j].low > previousOffsets[i].partitions[j].low ||
-                    newOffsets[i].partitions[j].high === newOffsets[i].partitions[j].low,
-                `Topic ${topics[i]} partition ${j} offset has not increased,
-                previousOffsets: ${previousOffsets[i].partitions[j].low} / ${previousOffsets[i].partitions[j].high},
-                newOffsets: ${newOffsets[i].partitions[j].low} / ${newOffsets[i].partitions[j].high}`);
+            // Checking topics offsets before kafkacleaner passes to be sure kafkacleaner works
+            // This function can be improved by consuming messages and
+            // verify that the timestamp is not older than last kafkacleaner run
+            // Instead of waiting for a fixed amount of time,
+            // we could also check for metrics to see last kafkacleaner run
+
+            // 10 seconds added to be sure kafkacleaner had time to process
+            await Utils.sleep(seconds * 1000 + 10000);
+
+            const newOffsets = await getTopicsOffsets(topics, kafkaAdmin);
+
+            for (let i = 0; i < topics.length; i++) {
+                process.stdout.write(`\nChecking topic ${topics[i]}\n`);
+                for (let j = 0; j < newOffsets[i].partitions.length; j++) {
+                    // Checking that the min offset has increased due to kafkacleaner
+                    // or that it didn't need to change because there was no new messages
+                    // or new messages came after a clean but still cleaned to previous high offset
+                    const lowOffsetIncreased = newOffsets[i].partitions[j].low > previousOffsets[i].partitions[j].low;
+                    const noNewMessages = newOffsets[i].partitions[j].high === newOffsets[i].partitions[j].low;
+                    const newMessagesAfterClean =
+                        newOffsets[i].partitions[j].low === previousOffsets[i].partitions[j].high &&
+                        previousOffsets[i].partitions[j].high !== '0';
+                    if (newMessagesAfterClean) {
+                        process.stdout.write(`New messages after clean for topic ${topics[i]} rechecking after`);
+                        continue;
+                    }
+                    assert.ok(lowOffsetIncreased || noNewMessages,
+                        `Topic ${topics[i]} partition ${j} offset has not increased,
+                        previousOffsets: ${previousOffsets[i].partitions[j].low} /\
+                         ${previousOffsets[i].partitions[j].high},
+                        newOffsets: ${newOffsets[i].partitions[j].low} / ${newOffsets[i].partitions[j].high}`);
+                    topics.splice(i, 1);
+                }
             }
         }
+
+        assert(topics.length === 0, `Topics ${topics.join(', ')} still have not been cleaned`);
```

Comment:
here we should fail if `newMessagesAfterClean`

Reply:
Maybe i missed something, but this is already what it does, while there is `newMessagesAfterClean` we continue (i.e we don't remove the topic from the list of topics to be checked) and here, if there is topics that remains in the array, it's that they haven't been cleaned and we print them to be able to debug in case it happens